docs/plugins/create.md

```diff
@@ -1,104 +1,525 @@
+If the `bin/download` script is absent, this script should download **and** install the specified version.
 
-If this script is not present asdf will assume that the `bin/install` script is present and will download and install the version. asdf only works without this script to support legacy plugins. All plugins must include this script, and eventually support for legacy plugins will be removed.
+For compatibility with versions of the asdf core earlier than `0.7._` and newer than `0.8._`, check for the presence of the `ASDF_DOWNLOAD_PATH` environment
+variable. If set, assume the `bin/download` script already downloaded the version, else download the source code in the `bin/install` script.
 
-#### bin/install
+**Environment Variables available to script**
 
-This script should install the version, in the path mentioned in `ASDF_INSTALL_PATH`. By default, asdf will create shims for any files in `$ASDF_INSTALL_PATH/bin` (this can be customized with the optional [bin/list-bin-paths](#binlist-bin-paths) script).
+- `ASDF_INSTALL_TYPE`: `version` or `ref`
+- `ASDF_INSTALL_VERSION`:
+  - Full version number if `ASDF_INSTALL_TYPE=version`.
+  - Git ref (tag/commit/branch) if `ASDF_INSTALL_TYPE=ref`.
+- `ASDF_INSTALL_PATH`: The path to where the tool _has been_, or _should be_ installed.
+- `ASDF_CONCURRENCY`: The number of cores to use when compiling source code. Useful for setting flags like `make -j`.
+- `ASDF_DOWNLOAD_PATH`: The path where the source code or binary was downloaded to.
 
-The install script should exit with a status of `0` when the installation is successful. If the installation fails the script should exit with any non-zero exit status.
+**Commands that invoke this script**
 
-If possible the script should only place files in the `ASDF_INSTALL_PATH` directory once the build and installation of the tool is deemed successful by the install script. asdf [checks for the existence](https://github.com/asdf-vm/asdf/blob/242d132afbf710fe3c7ec23c68cec7bdd2c78ab5/lib/utils.sh#L44) of the `ASDF_INSTALL_PATH` directory in order to determine if that version of the tool is installed. If the `ASDF_INSTALL_PATH` directory is populated at the beginning of the installation process other asdf commands run in other terminals during the installation may consider that version of the tool installed, even when it is not fully installed.
+- `asdf install`
+- `asdf install <tool>`
+- `asdf install <tool> [version]`
+- `asdf install <tool> latest[:version]`
+- `asdf install nodejs 18.0.0`: installs Node.js version `18.0.0` in the
+  `ASDF_INSTALL_PATH` directory.
 
-If you want your plugin to work with asdf version 0.7._ and earlier and version 0.8._ and newer check for the presence of the `ASDF_DOWNLOAD_PATH` environment variable. If it is not set download the source code in the bin/install callback. If it is set assume the `bin/download` script already downloaded it.
+**Call signature from asdf core**
+
+No parameters provided.
```

Comment:
Question: In Bash terminology, are values after the script name called "parameters" or "arguments" I would think they'd be called arguments in this case.

In a regular programming language, a function defines **parameters** it accepts, and when the function is invoked the caller passes it a set of corresponding **arguments**.